src/main/java/com/exactpro/th2/common/schema/message/impl/rabbitmq/connection/ConnectionManager.java

```diff
@@ -62,31 +63,30 @@
 
     private static final Logger LOGGER = LoggerFactory.getLogger(ConnectionManager.class);
 
-    private final Connection connection;
+    private static final String PUBLISH_CONNECTION_NAME = "Publish";
+    private static final String CONSUME_CONNECTION_NAME = "Consume";
+
+    private final Connection publishConnection;
+    private final Connection consumeConnection;
     private final Map<PinId, ChannelHolder> channelsByPin = new ConcurrentHashMap<>();
-    private final AtomicInteger connectionRecoveryAttempts = new AtomicInteger(0);
+    private final AtomicInteger publishConnectionRecoveryAttempts = new AtomicInteger(0);
+    private final AtomicInteger consumeConnectionRecoveryAttempts = new AtomicInteger(0);
     private final AtomicBoolean connectionIsClosed = new AtomicBoolean(false);
     private final ConnectionManagerConfiguration configuration;
     private final String subscriberName;
     private final AtomicInteger nextSubscriberId = new AtomicInteger(1);
-    private final ExecutorService sharedExecutor = Executors.newSingleThreadExecutor(new ThreadFactoryBuilder()
+    private final ExecutorService publishSharedExecutor = Executors.newSingleThreadExecutor(new ThreadFactoryBuilder()
+            .setNameFormat("rabbitmq-shared-pool-%d")
+            .build());
+    private final ExecutorService consumeSharedExecutor = Executors.newSingleThreadExecutor(new ThreadFactoryBuilder()
             .setNameFormat("rabbitmq-shared-pool-%d")
             .build());
 
-    private final HealthMetrics metrics = new HealthMetrics(this);
+    private final HealthMetrics publishMetrics = new HealthMetrics(this);
+    private final HealthMetrics consumeMetrics = new HealthMetrics(this);
 
-    private final RecoveryListener recoveryListener = new RecoveryListener() {
-        @Override
-        public void handleRecovery(Recoverable recoverable) {
-            LOGGER.debug("Count tries to recovery connection reset to 0");
-            connectionRecoveryAttempts.set(0);
-            metrics.getReadinessMonitor().enable();
-            LOGGER.debug("Set RabbitMQ readiness to true");
-        }
-
-        @Override
-        public void handleRecoveryStarted(Recoverable recoverable) {}
-    };
+    private final RecoveryListener publishRecoveryListener = getRecoveryListener(PUBLISH_CONNECTION_NAME);
+    private final RecoveryListener consumehRecoveryListener = getRecoveryListener(CONSUME_CONNECTION_NAME);
```

Comment:
Type in consume**h**RecoveryListener.
And maybe we don't need variables `publishRecoveryListener` and `consumehRecoveryListener` and we can use inline in `.addRecoveryListener()`?